pkg/operator/cassandra/controller/cluster.go

```diff
@@ -0,0 +1,69 @@
+package controller
+
+import (
+	"fmt"
+	cassandrav1alpha1 "github.com/rook/rook/pkg/apis/cassandra.rook.io/v1alpha1"
+	"github.com/rook/rook/pkg/operator/cassandra/controller/util"
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+)
+
+const (
+	StatefulSetName = "%s-%s-%s"
+)
+
+// UpdateStatus updates the status of the given Cassandra Cluster
+func (cc *ClusterController) UpdateStatus(c *cassandrav1alpha1.Cluster) error {
+
+	clusterStatus := cassandrav1alpha1.ClusterStatus{
+		Racks: map[string]cassandrav1alpha1.RackStatus{},
+	}
+	logger.Infof("Updating Status for cluster %s in namespace %s", c.Name, c.Namespace)
+
+	// Update readyMembers
+	for _, rack := range c.Spec.Datacenter.Racks {
+
+		status := cassandrav1alpha1.RackStatus{}
+
+		// Fetch corresponding StatefulSet from lister
+		sts, err := cc.statefulSetLister.StatefulSets(c.Namespace).
+			Get(fmt.Sprintf(StatefulSetName, c.Name, c.Spec.Datacenter.Name, rack.Name))
+		// If it wasn't found, continue
+		if errors.IsNotFound(err) {
+			continue
+		}
+		// If we got a different error, requeue and log it
+		if err != nil {
+			return err
+		}
+
+		// Update readyMembers
+		status.ReadyMembers = sts.Status.ReadyReplicas
+		clusterStatus.Racks[rack.Name] = status
+	}
+	c.Status = clusterStatus
```

Comment:
does this just update the status in memory? do we need to send the update to the k8s API server?

Reply:
I'm not entirely sure what way to explicitly consume the status subresource.  We may just be able to call the generated Update function for this type.  Your demo showed the status being updated over time, so how is that working?